recipes/haproxy_phpbb.rb

```diff
@@ -0,0 +1,62 @@
+#
+# Cookbook:: osl-keepalived
+# Recipe:: haproxy_phpbb
+#
+# Copyright:: 2018, Oregon State University
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+node.default['osl-keepalived']['virtual_ipv4'] = %w(140.211.15.244/24)
+node.default['osl-keepalived']['virtual_ipv6'] = %w()
+
+node.default['osl-keepalived']['master'] = {
+  'lb1.phpbb.org' => true,
+  'lb2.phpbb.org' => false
+}
+
+node.default['osl-keepalived']['priority'] = {
+  'lb1.phpbb.org' => 200,
+  'lb2.phpbb.org' => 100
+}
+
+include_recipe 'osl-keepalived::default'
+
+secrets = data_bag_item('osl_keepalived', 'haproxy_phpbb')
+
+keepalived_vrrp_instance 'haproxy-phpbb-ipv4' do
+  master node['osl-keepalived']['master'][node['fqdn']]
+  interface node['network']['default_interface']
+  virtual_router_id 1
+  priority node['osl-keepalived']['priority'][node['fqdn']]
+  authentication auth_type: 'PASS', auth_pass: secrets['auth_pass']
+  virtual_ipaddress node['osl-keepalived']['virtual_ipv4']
+  not_if { node['osl-keepalived']['virtual_ipv4'].empty? }
+end
+
+keepalived_vrrp_instance 'haproxy-phpbb-ipv6' do
+  master node['osl-keepalived']['master'][node['fqdn']]
+  interface node['network']['default_interface']
+  virtual_router_id 2
+  priority node['osl-keepalived']['priority'][node['fqdn']]
+  # Authentication isn't actually used with IPv6 (see https://tools.ietf.org/html/rfc5798#section-9)
+  # however, keepalived cookbook requires this to be defined
+  authentication auth_type: 'PASS', auth_pass: secrets['auth_pass']
+  virtual_ipaddress node['osl-keepalived']['virtual_ipv6']
+  not_if { node['osl-keepalived']['virtual_ipv6'].empty? }
+end
```

Comment:
Unless the phpBB project is going to start using IPv6 soon, I'm not sure if we should add this yet. I think it's OK to keep it here, but maybe ask them if they'd like to have it setup.

Reply:
Paul mentioned on IRC that they think they're interested.